langchain4j-vertex-ai/src/test/java/dev/langchain4j/model/vertexai/VertexAiEmbeddingModelIT.java

```diff
@@ -52,4 +53,111 @@ void testEmbeddingModel() {
 
         assertThat(response.finishReason()).isNull();
     }
+
+    @Test
+    void testTokensCountCalculationAndBatching() {
+        VertexAiEmbeddingModel model = VertexAiEmbeddingModel.builder()
+            .endpoint(System.getenv("GCP_VERTEXAI_ENDPOINT"))
+            .project(System.getenv("GCP_PROJECT_ID"))
+            .location(System.getenv("GCP_LOCATION"))
+            .publisher("google")
+            .modelName("textembedding-gecko@001")
+            .maxRetries(3)
+            .build();
+
+        List<Integer> tokenCounts = model.getTokensCounts(createRandomSegments(5000, 1000));
+
+        assertThat(tokenCounts).hasSize(5000);
+        assertThat(tokenCounts.stream().mapToInt(i -> i).sum()).isGreaterThanOrEqualTo(1_000_000);
```

Comment:
minor: please comment where `1_000_000` comes from

Reply:
OK, I can add a comment.
I think it's just that 5000 segments of 1000 characters is about 5 million characters.
And with an estimate of 5 letters per token, we have at least 1 million tokens in total.
It's a bit arbitrary.

Reply:
Comment added